### PR TITLE
Added reconnect method

### PR DESCRIPTION
**Pull request by tdl-jturner**
Solutions for issue #49 and #43 

Adding a reconnect to Bot allows the following (or similar) to be implemented to reconnect on closure:

    @Override
    public void afterConnectionClosed(WebSocketSession session, CloseStatus status) {
        logger.debug("WebSocket closed: {}, Close Status: {}", session, status.toString());
        this.reconnect();
        logger.debug("WebSocket Reconnected: {}", session);
    }

### CODE REVIEW

**Comment by rampatra** [comment]
If you `reconnect()` like this then it will never let you end the socket connection if you ever wanted to. In fact, you can override this method in your bot and reconnect under specific cases.

**Comment by tdl-jturner** [comment]
Thank you for the response. I agree that the use of the reconnect should be the user's choice and thus why the override is in the implementation and not the base class. This was just an example usage of the new reconnect procedure. The reconnect needs to be in the base class as it is references multiple private procedures.

**Comment by rampatra** [comment]
Sorry, my bad. I assumed the above code snippet is in your PR. I will merge your PR now. Thanks for contributing 👍

File: jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java [hunk 1]
```diff
@@ -292,4 +292,17 @@ private void startWebSocketConnection() {
             logger.error("No websocket url returned by Slack.");
         }
     }
+
+    /**
+     * Method to reconnect to websocket
+     *
+     */
+    public void reconnect() {
+    	if (slackService.getWebSocketUrl() != null) {
```
Comment (rampatra) [inline jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java]: Code is repeated. Maybe we can pull out the common code from `startWebSocketConnection` and `reconnect` methods to another method.

Comment (tdl-jturner) [reply jbot/src/main/java/me/ramswaroop/jbot/core/slack/Bot.java]: That's a good point. Renamed it to a public openConnection method and removed duplicate code. Now it is logically named for both locations.

**Comment by tdl-jturner** [comment]
Thank you!

**Pull request merged**